torch/optim/adadelta.py

```diff
@@ -79,7 +81,11 @@ def step(self, closure=None):
             grads = []
             square_avgs = []
             acc_deltas = []
-            lr, rho, eps, weight_decay = group['lr'], group['rho'], group['eps'], group['weight_decay']
+            lr, rho, eps, weight_decay, foreach = (group['lr'],
+                                                   group['rho'],
+                                                   group['eps'],
+                                                   group['weight_decay'],
+                                                   group['foreach'])
```

Comment:
Sorry I missed that in previous reviews...
We want to make sure that checkpoint saved with an older version of pytorch still work here.
I think the simplest way to do that is to add to this class:
```python
def __setstate__(self, state):
    super().__setstate__(state)
    for group in self.param_groups:
        if 'forearch' not in group:
            group['foreach'] = False
```

If you want to test this. You can create an instance of this optimizer. Get the state dict from it. Delete the foeach entries in it and then load_state_dict.
And you should still be able to step with that optimizer after loading this state dict.

Reply:
Fixed!